packages/build/tests/plugins/tests.js

```diff
@@ -1040,39 +1040,77 @@ test('Does not pin netlify.toml-only plugin versions if there are no site ID', a
+  })
+})
+
+test('No --node-path version validation is performed if the plugins_system_node_version feature flag is set', async (t) => {
+  const nodePath = getNodePath('8.2.0')
+  await runFixture(t, 'node_version_simple', {
+    flags: { nodePath, featureFlags: { buildbot_build_plugins_system_node_version: true } },
+  })
 })
 
 test('Validate --node-path unsupported version does not fail when no plugins are used', async (t) => {
   const nodePath = getNodePath('8.2.0')
-  await runFixture(t, 'empty', { flags: { nodePath } })
+  await runFixture(t, 'empty', {
+    flags: { nodePath, featureFlags: { buildbot_build_plugins_system_node_version: false } },
+  })
 })
 
 test('Validate --node-path version is supported by the plugin', async (t) => {
   const nodePath = getNodePath('9.0.0')
-  await runFixture(t, 'engines', { flags: { nodePath } })
+  await runFixture(t, 'engines', {
+    flags: { nodePath, featureFlags: { buildbot_build_plugins_system_node_version: false } },
+  })
+})
+
+test('No --node-path version validation is performed against the plugin engines if the plugins_system_node_version feature flag is set', async (t) => {
+  const nodePath = getNodePath('9.0.0')
+  await runFixture(t, 'engines', {
+    flags: { nodePath, featureFlags: { buildbot_build_plugins_system_node_version: true } },
+  })
 })
 
 test('Validate --node-path', async (t) => {
-  await runFixture(t, 'node_version_simple', { flags: { nodePath: '/doesNotExist' } })
+  await runFixture(t, 'node_version_simple', {
+    flags: { nodePath: '/doesNotExist', featureFlags: { buildbot_build_plugins_system_node_version: false } },
+  })
+})
+
+test('Provided --node-path version produces a node version warning if the build is being run in buildbot with local plugins and the plugins_system_node_version ff is set to false', async (t) => {
+  const nodePath = getNodePath('9.0.0')
+  await runFixture(t, 'engines', {
+    flags: { nodePath, mode: 'buildbot', featureFlags: { buildbot_build_plugins_system_node_version: false } },
+  })
 })
 
-test('Provided --node-path version produces a node version warning if the build is being run in buildbot with local plugins', async (t) => {
+test('Provided --node-path version has no effect for plugin executions when the feature flag plugins_system_node_version is set', async (t) => {
   const nodePath = getNodePath('9.0.0')
-  await runFixture(t, 'engines', { flags: { nodePath, mode: 'buildbot' } })
+  await runFixture(t, 'engines', {
+    flags: { nodePath, mode: 'buildbot', featureFlags: { buildbot_build_plugins_system_node_version: true } },
+  })
 })
 
 test('Provided --node-path version does not produce a node version warning if the build is being run locally', async (t) => {
   const nodePath = getNodePath('9.0.0')
-  await runFixture(t, 'engines', { flags: { nodePath, mode: 'cli' } })
+  await runFixture(t, 'engines', {
+    flags: { nodePath, mode: 'cli', featureFlags: { buildbot_build_plugins_system_node_version: false } },
```

Comment:
In the future, we can migrate some of those `--node-path` tests to running a build command instead of a build plugin.